cleanup_utils/dns_cleanup.py

```diff
@@ -79,6 +86,37 @@ def get_dns_record_list(self, zone_id):
 
         return dns_entries
 
+    def delete_dns_record(self, zone_id, zone_version_id, record_name):
+        """
+        Delete a record from a version of the domain
+        """
+        # Delete record
+        try:
+            self.client_zone.record.delete(self.api_key, zone_id, zone_version_id, {
+                'type': ['A', 'CNAME'],
+                'name': record_name,
+            })
+        except xmlrpc.client.Fault as e:
+            logger.error("  > FAILED Deleting DNS entries for %s...", record_name)
+            logger.error("  > ERROR: %s", e)
+
+    def create_new_zone_version(self, zone_id):
+        """
+        Create a new version of the domain, based on the current version
+        Returns the `version_id` of the version
+        """
+        return self.client_zone.version.new(self.api_key, zone_id)
```

Comment:
@giovannicimolin, in case of concurrent operations, can this become stale before the operations that use this actually run? If that is possible, can we handle the error and retry?

Reply:
This creates a new zone version and deletes all cleaned up hashes on that version. 
Then, only after completing all operations, `self.set_zone_version` is ran, so, if two cleanup scripts run concurrently, each of them will create a new zone and the one that finishes last will set the domains DNS records.
There's no way to fix this unless we either run this separately (outside the CI) or make some kind of atomic operation for all the cleanup runs.
Since this PR also introduces workflows to avoid running multiple cleanup instances at the same time we shouldn't have issues with this. Also, the entire cleanup run in pretty fast (<3minutes), so it's gonna be somewhat hard to have a concurrent execution.

Reply:
Also, between putting this outside the CI and making it an "atomic transaction" between all concurrent instances, I think that hosting this outside the CI is the best option. 
However, both of these options are more complex that the solution that we have today, and might be a huge over engineering. 
But there's a plus side: if this is hosted outside the CI, we can make this cleanup tool watch over all Opencraft's resources and prune them according to our necessities, not only the integration resources.

Reply:
@lgp171188

Reply:
@giovannicimolin, the only rare chance for concurrency issues is when some one edits a zone while the CI checks are running. But a re-run should probably fix it. So I guess we can live with that.